### PR TITLE
only install cfssl/cfssljon from vendor if they are not in path

### LINKED ISSUE #67286: Unable to install 'cfssl' from vendor
>
/kind bug

**What happened**:
Unable to spin up cluster using the kubetest, with 'ci/latest' image.
The CI jobs are failing, because the cluster is not provisioned properly.

```
I0810 23:40:23.527] Installing cfssl from vendor
I0810 23:40:23.548] Checking for custom logdump instances, if any
I0810 23:40:23.552] Sourcing kube-util.sh
I0810 23:40:23.587] Detecting project
I0810 23:40:23.588] Project: k8s-jkns-gke-reboot-1-4
I0810 23:40:23.588] Network Project: k8s-jkns-gke-reboot-1-4
I0810 23:40:23.588] Zone: us-central1-b
I0810 23:40:23.588] Dumping logs from master locally to '/workspace/_artifacts'
W0810 23:40:23.689] can't load package: package k8s.io/kubernetes/vendor/github.com/cloudflare/cfssl/cmd/cfssl: cannot find package "k8s.io/kubernetes/vendor/github.com/cloudflare/cfssl/cmd/cfssl" in any of:
W0810 23:40:23.689] 	/usr/local/go/src/k8s.io/kubernetes/vendor/github.com/cloudflare/cfssl/cmd/cfssl (from $GOROOT)
W0810 23:40:23.690] 	/go/src/k8s.io/kubernetes/vendor/github.com/cloudflare/cfssl/cmd/cfssl (from $GOPATH)
W0810 23:40:23.690] 2018/08/10 23:40:23 process.go:155: Step './hack/e2e-internal/e2e-up.sh' finished in 1m26.617131334s

```

I have tried locally, i am still getting the same error.
```
Write out database with 1 new entries
Data Base Updated
+ kube::util::ensure-cfssl /tmp/tmp.CzcIHwWJK4/cfssl
+ echo 'Installing cfssl from vendor'
Installing cfssl from vendor
+ GOBIN=
+ go install k8s.io/kubernetes/vendor/github.com/cloudflare/cfssl/cmd/cfssl
can't load package: package k8s.io/kubernetes/vendor/github.com/cloudflare/cfssl/cmd/cfssl: cannot find package "k8s.io/kubernetes/vendor/github.com/cloudflare/cfssl/cmd/cfssl" in any of:
        /usr/local/go/src/k8s.io/kubernetes/vendor/github.com/cloudflare/cfssl/cmd/cfssl (from $GOROOT)
        /home/ubuntu/e2epath/src/k8s.io/kubernetes/vendor/github.com/cloudflare/cfssl/cmd/cfssl (from $GOPATH)
+ GOBIN=
+ go install k8s.io/kubernetes/vendor/github.com/cloudflare/cfssl/cmd/cfssljson
can't load package: package k8s.io/kubernetes/vendor/github.com/cloudflare/cfssl/cmd/cfssljson: cannot find package "k8s.io/kubernetes/vendor/github.com/cloudflare/cfssl/cmd/cfssljson" in any of:
        /usr/local/go/src/k8s.io/kubernetes/vendor/github.com/cloudflare/cfssl/cmd/cfssljson (from $GOROOT)
        /home/ubuntu/e2epath/src/k8s.io/kubernetes/vendor/github.com/cloudflare/cfssl/cmd/cfssljson (from $GOPATH)
++ PATH=:/home/ubuntu/bin:/home/ubuntu/.local/bin:/home/ubuntu/y/google-cloud-sdk/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/usr/local/go/bin:/home/ubuntu/e2epath/bin:GOPATH=/home/ubuntu/e2epath
++ command -v cfssl
+ CFSSL_BIN=
++ PATH=:/home/ubuntu/bin:/home/ubuntu/.local/bin:/home/ubuntu/y/google-cloud-sdk/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/usr/local/go/bin:/home/ubuntu/e2epath/bin:GOPATH=/home/ubuntu/e2epath
++ command -v cfssljson
+ CFSSLJSON_BIN=
+ [[ ! -x '' ]]
+ echo 'Failed to install cfssl.'
Failed to install cfssl.
+ exit 1

```

[Refer PR-128](https://github.com/kubernetes-sigs/poseidon/pull/128)
[Build-logs](https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_poseidon/128/ci-poseidon-e2e-gce/301/build-log.txt)

Could be releated to this PR #67178 

@BenTheElder @cblecker

/cc @deepak-vij

### PR DESCRIPTION
**Pull request by BenTheElder**
**What this PR does / why we need it**: ensures we only install cfssl/cfssljson from vendor if they are not already in the path, so the scripts can continue to work when used from releases

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #67286

**Special notes for your reviewer**: this fixes some CI tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by ixdy — COMMENTED** [review]
/lgtm

File: hack/lib/util.sh [hunk 1]
```diff
@@ -697,7 +697,7 @@ function kube::util::join {
   echo "$*"
 }
 
-# Downloads cfssl/cfssljson into $1 directory if they do not already exist in PATH
+# installs cfssl/cfssljson into $KUBE_OUTPUT_BINPATH if they do not already exist in PATH
 #
 # Assumed vars:
 #   $1 (cfssl directory) (optional)
```
Comment (ixdy) [inline hack/lib/util.sh]: remove this?

**Comment by cblecker** [comment]
/hold
The thing I *don't* like about this is that we can't ensure a consistent version. Considering this utility generates certs, security fixes are important.

Thoughts?

**Comment by BenTheElder** [comment]
> The thing I don't like about this is that we can't ensure a consistent version. Considering this utility generates certs, security fixes are important.

Do you mean between the two binaries, or between the repo and what we use?

If the former, I'm happy to do both or none. If the latter then the binaries need to be in our release, because otherwise you can't run the cluster scripts from a release tarball.

**Comment by ixdy** [comment]
/lgtm cancel

discussing better alternatives on slack, though probably starting with a quick rollback of just 7a10073e4aefbda1d0e026353fd055db5cf8f140.

**Comment by BenTheElder** [comment]
Agreed.

**Pull request closed without merging**